src/edit.js

```diff
@@ -35,48 +58,64 @@ export default function Edit( { attributes, setAttributes } ) {
 
 	useEffect( () => {
 		async function getRelatedPosts() {
-			const postId = window.location.href.match( /post=([\d]+)/ )[ 1 ];
-			const response = await window.fetch( `/wp-json/wp/v2/posts?search=${ countries[ countryCode ] }&exclude=${ postId }` );
-
-			if ( ! response.ok )
-				throw new Error( `HTTP error! Status: ${ response.status }` );
-
-			const posts = await response.json();
+			const posts = await apiFetch( {
+				path: `/wp/v2/posts?search=${ countries[ countryCode ] }&exclude=${ postId }`,
+			} ).then( ( response ) => {
+				return response;
+			} );
```

Comment:
@cdevroe Three questions here:

1. What are the benefits of using `apiFetch` compared to `window.fetch`?
2. Is there a more robust way to fetch data than `apiFetch`?
3. How could the related posts fetching be optimized even more (in terms of payload size)?

Reply:
> 1. What are the benefits of using `apiFetch` compared to `window.fetch`?

When I was looking into how to fix the REST API call to fetch the related posts I saw apiFetch was officially supported, regularly updated, etc. It seems to have some nice features specific to WordPress such as dealing with nonce when needed.

> 2. Is there a more robust way to fetch data than `apiFetch`?

There are other third-party clients that could be used to fetch data. apiFetch allows for custom fetch handlers that have better error handling, security features built-in, etc. I didn’t think it was warranted to bring in an entirely different fetch handler since we are calling from the local WordPress REST API. If I were pulling data from an external API, especially one that could potentially return security problems, I’d use something more robust.

> 3. How could the related posts fetching be optimized even more (in terms of payload size)?

I see two ways to reduce the size of the payload. One is to return only posts (on the off chance other types would be returned) and also using the `_fields` argument to limit the number of fields returned to only that which we will use. I’ve added this to the PR.